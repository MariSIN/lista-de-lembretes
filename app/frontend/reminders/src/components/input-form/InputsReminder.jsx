import { useContext } from 'react';
import InputMask from 'react-input-mask';
import Context from '../../context/Context';
import '../../style/inputsReminder.css';
import ButtonPostReminder from './ButtonPostReminder';

export default function InputsReminder() {
	const { nameDate, setDateName } = useContext(Context);
	const handleChange = ({ target }) => {
		const { name, value } = target;

		setDateName({
			...nameDate,
			[name]: value,
		});
	};

	return (
		<form className='form-container'>
			<h1>Novo Lembrete</h1>
			<div className='input-label-container'>
				<div className='input-label'>
					<label htmlFor='name-box'>Nome</label>
					<input
						id='name-box'
						type='text'
						name='name'
						value={nameDate.name}
						onChange={handleChange}
						placeholder='Nome do lembrete'
						required
					/>
				</div>
				<div className='input-label'>
					<label htmlFor='date-box'>Data</label>
					<InputMask
						id='date-box'
						mask='99/99/9999'
						placeholder='DD/MM/YYYY'
						name='date'
						value={nameDate.date}
						onChange={handleChange}
						required
					/>
				</div>
			</div>

			<ButtonPostReminder />
		</form>
	);
}
