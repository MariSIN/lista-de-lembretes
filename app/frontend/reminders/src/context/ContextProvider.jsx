import PropTypes from 'prop-types';
import { useMemo, useState } from 'react';
import Context from './Context';

function ContextProvider({ children }) {
	const [nameDate, setDateName] = useState({
		name: '',
		date: '',
	});
  const [reminder, setReminder] = useState({})

	const value = useMemo(
		() => ({
			nameDate,
			setDateName,
      reminder,
      setReminder
		}),
		[nameDate, reminder],
	);

	return <Context.Provider value={value}>{children}</Context.Provider>;
}

ContextProvider.propTypes = {
	children: PropTypes.node.isRequired,
};

export default ContextProvider;
