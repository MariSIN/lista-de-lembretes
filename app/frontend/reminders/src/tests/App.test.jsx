import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import InputsReminder from '../components/input-form/InputsReminder';
import ContextProvider from '../context/ContextProvider';

describe('Testa componente InputsReminder', () => {
  it('Testa se inputs Nome e Data existem', () => {
    render(
      <ContextProvider >
        <InputsReminder />
      </ContextProvider>
    )
    expect(screen.getByPlaceholderText('Nome do lembrete')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('DD/MM/YYYY')).toBeInTheDocument();
  });

  it('Testa se é possível escrever nos inputs', () => {
    render(
      <ContextProvider >
        <InputsReminder />
      </ContextProvider>
    )
    const inputName = screen.getByPlaceholderText('Nome do lembrete');
    const inputDate = screen.getByPlaceholderText('DD/MM/YYYY');
    userEvent.type(inputName, 'Lembrete 1');
    userEvent.type(inputDate, '20/10/2023');
  });
  it('Testa se o estado name é atualizado corretamente', () => {
    const { container } = render(
      <ContextProvider>
        <InputsReminder />
      </ContextProvider>
    );
    const inputName = screen.getByPlaceholderText('Nome do lembrete');
    userEvent.type(inputName, 'Novo Nome');

    const inputDate = screen.getByPlaceholderText('DD/MM/YYYY');
    userEvent.type(inputDate, '20/10/2023');

    expect(container).toMatchSnapshot();
  });
  
  
});
