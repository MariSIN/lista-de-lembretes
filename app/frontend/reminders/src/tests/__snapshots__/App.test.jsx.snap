// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa componente InputsReminder Testa se o estado name Ã© atualizado corretamente 1`] = `
<div>
   
  <form>
    <h1>
      Novo Lembrete
    </h1>
    <div>
      <label
        for="name-box"
      >
        Nome
      </label>
      <input
        id="name-box"
        name="name"
        placeholder="Nome do lembrete"
        required=""
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="date-box"
      >
        Data
      </label>
      <input
        id="date-box"
        name="date"
        placeholder="DD/MM/YYYY"
        required=""
        value=""
      />
    </div>
    <button
      type="button"
    >
      CRIAR
    </button>
  </form>
</div>
`;
